pymc3/distributions/multivariate.py

```diff
@@ -524,6 +518,38 @@ def logp(value, n, p):
         )
 
 
+class DirichletMultinomialRV(RandomVariable):
+    name = "dirichlet_multinomial"
+    ndim_supp = 1
+    ndims_params = [0, 1]
+    dtype = "int64"
+    _print_name = ("DirichletMN", "\\operatorname{DirichletMN}")
+
+    @classmethod
+    def rng_fn(cls, rng, n, a, size):
+
+        if n.ndim > 0 or a.ndim > 1:
+            n, a = broadcast_params([n, a], cls.ndims_params)
+            size = tuple(size or ())
+
+            if size:
+                n = np.broadcast_to(n, size + n.shape)
+                a = np.broadcast_to(a, size + a.shape)
+
+            res = np.empty(a.shape)
+            for idx in np.ndindex(a.shape[:-1]):
+                p = rng.dirichlet(a[idx])
+                res[idx] = rng.multinomial(n[idx], p)
+            return res
+        else:
+            # n is a scalar, a is a 1d array
+            p = rng.dirichlet(a)
+            return rng.multinomial(n, p, size=size)
```

Comment:
Does this look better @ricardoV94 ?

Reply:
I don't know where exactly is the problem (if there's any) but the sampling seems off.

```python
dirichlet_multinomial(5, np.ones(4)/100, size=10).eval()
array([[0, 0, 5, 0],
       [0, 0, 5, 0],
       [0, 0, 5, 0],
       [0, 0, 5, 0],
       [0, 0, 5, 0],
       [0, 0, 5, 0],
       [0, 0, 5, 0],
       [0, 0, 5, 0],
       [0, 0, 5, 0],
       [0, 0, 5, 0]])
```
I would expect more variation over rows. Here is what comes out from the V3 version
```python
pm.DirichletMultinomial.dist(n=5, a=np.ones(4)/100, shape=(4,)).random(size=10) 
array([[0., 5., 0., 0.],
       [0., 0., 0., 5.],
       [0., 0., 5., 0.],
       [0., 5., 0., 0.],
       [0., 5., 0., 0.],
       [0., 0., 5., 0.],
       [0., 0., 5., 0.],
       [1., 0., 0., 4.],
       [5., 0., 0., 0.],
       [0., 0., 5., 0.]])
```

Reply:
Thanks, that was very helpful! I fixed this implementation and it seems to work well now:
```python
rng = aesara.shared(np.random.RandomState(123), borrow=True)

# 1d case 
x = pm.DirichletMultinomial.dist(5, np.ones(4)/100, size=10, rng=rng)
x.eval()
array([[0, 0, 0, 5],
       [5, 0, 0, 0],
       [0, 0, 0, 5],
       [5, 0, 0, 0],
       [0, 0, 0, 5],
       [0, 0, 5, 0],
       [0, 0, 5, 0],
       [0, 5, 0, 0],
       [5, 0, 0, 0],
       [5, 0, 0, 0]])

# multidim case
x = pm.DirichletMultinomial.dist(np.array([5, 10]), np.array([np.ones(4)/100, np.array([1, 1, 1, 100])]), size=10, rng=rng)
x.eval()
array([[[ 4,  0,  1,  0],
        [ 0,  1,  0,  9]],

       [[ 0,  0,  5,  0],
        [ 0,  0,  0, 10]],

       [[ 0,  0,  0,  5],
        [ 0,  1,  0,  9]],

       [[ 0,  0,  0,  5],
        [ 1,  0,  0,  9]],

       [[ 0,  0,  0,  5],
        [ 0,  0,  0, 10]],

       [[ 0,  0,  0,  5],
        [ 0,  0,  0, 10]],

       [[ 0,  0,  0,  5],
        [ 0,  0,  0, 10]],

       [[ 5,  0,  0,  0],
        [ 0,  0,  0, 10]],

       [[ 0,  5,  0,  0],
        [ 0,  0,  0, 10]],

       [[ 0,  0,  0,  5],
        [ 1,  0,  1,  8]]])